libs/vkd3d/device.c

```diff
@@ -1640,6 +1644,12 @@ static HRESULT vkd3d_select_physical_device(struct vkd3d_instance *instance,
         if (i == device_index)
             device = physical_devices[i];
 
+        if (!strstr(device_properties.deviceName, filter))
```

Comment:
Is it well defined how strstr works with empty string? Seems cleaner to check `if (filter && strstr())`.

Reply:
Yes, it's well defined (it will return the first argument). But yes, `if (filter && !strstr()).` is probably cleaner.